app/src/main/java/com/github/mateo762/myapplication/search/SearchActivity.kt

```diff
@@ -30,18 +36,37 @@ class SearchActivity : BaseActivity() {
         layoutManager.orientation = LinearLayoutManager.VERTICAL
         binding.recyclerView.layoutManager = layoutManager
 
-        val items = listOf(
-            SearchItem("Pedro", "Son of Harry"),
-            SearchItem("Alice", "Married to Bob"),
-            SearchItem("Peter", "Married to Mary"),
-            SearchItem("Elisabeth", "Mother of Pascal"),
-            SearchItem("Charlotte", "Daughter of Boris")
-        )
+        println("Loaded the Search Activity")
 
-        adapter = SearchViewAdapter(items)
+        //TODO: Retrieve a list of users from Firebase
+        val database = FirebaseDatabase.getInstance()
+        val usersRef = database.getReference("usernames")
 
+        val users = mutableListOf(SearchItem("User", "Description"))
+        adapter = SearchViewAdapter(users)
         binding.recyclerView.adapter = adapter
 
```

Comment:
So, if I'm getting it right, here we display all the SearchItem available for this user right?

Reply:
Yeah exactly. In this case I had to call adapter and binding here, and then recall it within the OnDataChanged because I couldn't use the adapter.NotifyOnDataChanged (it didn't update the recycler view for some reason, could be because of binding use?)